modules/styling/lib/cs.ts

```diff
@@ -722,25 +730,7 @@ export function handleCsProp<
   // If runtime styles are detected, we need to do extra work to ensure styles merge according to
   // the rules of Emotion's runtime.
   if (shouldRuntimeMergeStyles) {
-    const registeredStyles: string[] = [];
-
-    // We are using the raw `css` instead of `createStyles` because `css` uses style hashing and
-    // `createStyles` generates a new ID every time. We could use `createStyles` here, but it would
-    // be more wasteful and new styles would be generated each React render cycle. `css` is safe to
-    // use inside a render function while `createStyles` should always be used outside the React
-    // render cycle. The following is `@emotion/utils` and it mutates the passed in
-    // `registeredStyles` array. It uses the cache of registered styles and the original className.
-    // It returns a string with found registered class names removed.
-    returnProps.className = getRegisteredStyles(
-      cache.registered,
-      registeredStyles,
-      returnProps.className || ''
-    );
-
-    // The `className` is mutated again, but with a brand new CSS class name of all the merged
-    // registered styles. This is how Emotion merges styles from the `css` prop and `styled`
-    // components.
-    returnProps.className += css(registeredStyles);
```

Comment:
I didn't realize before that the `cx` function from `@emotion/css` does this exact thing. All I need to do is detect if there are runtime dynamic styles and call the `cx` function from the Emotion instance if there is and Emotion will handle style merging for me. This reduces the code we maintain and have to explain and still passes all our tests.